.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: Pre-commit checks
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
We do not need all logs etc in YUNIKORN-285 we changed the fetch depth to 2 to make it all work and it safes a lot of data being pulled out.
We should have `fetch-depth: 2`, will fix on commit